src/components/SingleTopic.jsx

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useState } from "react";
```

Comment:
This component does not need to exist. You've made a clever api call that can check for a topic, consider drying up your components by re using all articles to handle both articles and articles by topic.